### DISCUSSION TITLE
Configure a CDN, Restrict Public Access to S3

### DISCUSSION THREAD

**Discussion by centaurreader** [architect/architect#1288] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
Granting public access to objects hosted in an S3 bucket results in warning emails from Amazon as well as warnings within the S3 control panel. Amazon suggests that you never grant public access in their documentation and in the settings. (Note: they do state, "If you have a business need to maintain some level of public access, please see Overview of Managing Access (https://docs.aws.amazon.com/AmazonS3/latest/dev/access-control-overview.html#access-control-resources-manage-permissions-basics) for more in-depth instructions on managing access to your bucket to make sure you’ve permitted the correct level of access to your objects."

**Describe the solution you'd like**
I'd like to allow public access via a CDN (Cloudfront?) and to secure my S3 buckets so that public access is prevented and the AWS warnings quiet down.

**Describe alternatives you've considered**
Technically, there is nothing wrong here. Public access is currently limited to "List" only, however, the warnings in the panel and emails lead me, as a user, to feel I've done something wrong and to doubt my setup.

**Comment by exalted** [comment]
Man, I like how plain and simple and humble you've written this issue @centaurreader! 🙇 & 👏

**Comment by ryanblock** [comment]
Converted to discussion to discuss more here about what we may want to do with CDNs. Right now it's a bit in-between for Architect, and could probably use some more baking (especially accounting for some AWS features we expect coming in 2022).
